pkg/controller/deployment/deployment_controller.go

```diff
@@ -1294,3 +1413,27 @@ func (dc *DeploymentController) rollbackToTemplate(deployment *extensions.Deploy
 	d, err = dc.updateDeploymentAndClearRollbackTo(deployment)
 	return
 }
+
+// isScalingEvent checks whether the provided deployment has been updated with a scaling event
+// by looking at the desired-replicas annotation in the active replica sets of the deployment.
+func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) bool {
+	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
+	if err != nil {
+		return false
+	}
+	allRSs := append(oldRSs, newRS)
+	for _, rs := range controller.FilterActiveReplicaSets(allRSs) {
+		if desiredString, ok := rs.Annotations[deploymentutil.DesiredReplicasAnnotation]; ok {
```

Comment:
The deployment controller (and any other controller loop for that matter) is a client to the apiserver and watches etcd for changes. Bob creates a deployment, it is stored on etcd, a watch event is triggered, the controller reconciles the deployment and tries to fulfill its spec (first rollout). Bob edits the podtemplate of his deployment, etcd, watch event, the controller start a new rollout. During the rollout, the controller updates annotations and status fields on the deployment as a result of updating replica sets that need to be reconciled, which will make the deployment come back as an event. Bob scales up in between, another watch event for the controller to handle. The deployment will be enqueued for all watch events. The controller should be able to recognize scaling events.